salt/recommendations/init.sls

```diff
@@ -99,8 +99,33 @@ recommendations-docker-compose:
         - require:
             - file: recommendations-docker-compose
 
+{% if pillar.elife.webserver.app == "caddy" %}
 
-recommendations-nginx-vhost:
+recommendations-caddy-requisite:
+    cmd.run:
+        - name: |
+            set -e
+            rm -rf /srv/recommendations/web
+            mkdir /srv/recommendations/web
+            touch /srv/recommendations/web/app.php
+        - runas: {{ pillar.elife.deploy_user.username }}
+        - require:
+            - recommendations-folder
```

Comment:
the annotations formula suffers from this as well and solves it a little more nicely:

```
annotations-folder-web:
    file.managed:
        - name: /srv/annotations/web/app.php
        - contents: '# placeholder. Nginx requires this file to pass requests to a php-fpm container'
        - makedirs: True
        - require:
            - annotations-folder
```